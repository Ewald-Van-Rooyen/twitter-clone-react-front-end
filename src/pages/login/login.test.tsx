import TestRenderer, {ReactTestRenderer} from "react-test-renderer";

import React from "react";
import LoginPage from "./loginPage";
import {fireEvent, render, screen, wait, waitFor} from '@testing-library/react'
import userEvent from '@testing-library/user-event'

let container: any;
let testRenderer: ReactTestRenderer;

describe("Login component tests", () => {

    it("should render the Login component", () => {
        testRenderer = TestRenderer.create(
            <LoginPage onSubmitClickCallback={() => {
            }}/>
        );
        expect(testRenderer).toMatchSnapshot();
    });

    test('rendering and submitting a basic Formik form failing vallidation', async () => {
        const {container} = render(<LoginPage onSubmitClickCallback={() => {
        }}/>);


        const username = container.querySelector('input[name="username"]');
        const password = container.querySelector('input[name="password"]');
        const submit = container.querySelector('button[type="submit"]');

        await waitFor(() => {
            fireEvent.change(username, {
                target: {
                    value: ""
                }
            })
        });

        await waitFor(() => {
            fireEvent.change(password, {
                target: {
                    value: ""
                }
            })
        });

        await waitFor(() => {
            fireEvent.click(submit);
        });

        const passwordErrorMessage = container.querySelector("#passwordErrorMessage");
        const usernameErrorMessage = container.querySelector("#usernameErrorMessage");

        expect(passwordErrorMessage.innerHTML).toBe("Password is required");
        expect(usernameErrorMessage.innerHTML).toBe("Username is required");
    });

    test("rendering and submitting a basic Formik form failing password length", async () => {
        const {container} = render(<LoginPage onSubmitClickCallback={() => {
        }}/>);


        const username = container.querySelector('input[name="username"]');
        const password = container.querySelector('input[name="password"]');
        const submit = container.querySelector('button[type="submit"]');

        await waitFor(() => {
            fireEvent.change(username, {
                target: {
                    value: "123456"
                }
            })
        });

        await waitFor(() => {
            fireEvent.change(password, {
                target: {
                    value: "124"
                }
            })
        });

        await waitFor(() => {
            fireEvent.click(submit);
        });

        const passwordErrorMessage = container.querySelector("#passwordErrorMessage");
        expect(passwordErrorMessage.innerHTML).toBe("Password should be of minimum 4 characters length");
    });

});
