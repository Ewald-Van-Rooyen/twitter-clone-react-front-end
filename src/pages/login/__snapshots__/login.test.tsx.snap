// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component tests should render the Login component 1`] = `
<main
  className="main"
>
  <div
    className="button-twitter"
  />
  <form
    className="login"
    onSubmit={[Function]}
  >
    <svg
      className="login-sides"
    >
      <line
        className="top-right first"
        x1="50%"
        x2="100%"
        y1="0"
        y2="0"
      />
      <line
        className="top-left first"
        x1="50%"
        x2="0"
        y1="0"
        y2="0"
      />
      <line
        className="right second"
        x1="100%"
        x2="100%"
        y1="0"
        y2="100%"
      />
      <line
        className="left second"
        x1="0"
        x2="0"
        y1="0"
        y2="100%"
      />
      <line
        className="bottom-left third"
        x1="0"
        x2="50%"
        y1="100%"
        y2="100%"
      />
      <line
        className="bottom-right third"
        x1="100%"
        x2="50%"
        y1="100%"
        y2="100%"
      />
    </svg>
    <fieldset
      className="login-fieldset"
    >
      <input
        className="login-fieldset-field"
        name="username"
        onChange={[Function]}
        placeholder="Username"
        type="text"
        value=""
      />
      <input
        className="login-fieldset-field"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
      <button
        className="login-fieldset-submit"
        type="submit"
      >
        Log In
      </button>
    </fieldset>
  </form>
</main>
`;
